src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -72,6 +88,71 @@
      */
     private boolean expired = false;
 
+    /**
+     * The set of IDs of the pinned endpoints of this {@code Endpoint}.
+     */
+    private Set<String> pinnedEndpoints = new HashSet<>();
+
+    /**
+     * The set of currently selected <tt>Endpoint</tt>s at this
+     * <tt>Endpoint</tt>.
+     */
+    private Set<String> selectedEndpoints = new HashSet<>();
+
+
+    /**
+     * Sets the list of pinned endpoints for this endpoint.
+     * @param newPinnedEndpoints the set of pinned endpoints.
+     * @return true if the underlying set of pinned endpoints has changed, false
+     * otherwise. The return value was introduced to enable overrides to
+     * act upon the underlying set changing.
+     */
+    public void pinnedEndpointsChanged(Set<String> newPinnedEndpoints)
+    {
+        // Check if that's different to what we think the pinned endpoints are.
+        Set<String> oldPinnedEndpoints = this.pinnedEndpoints;
+        if (!oldPinnedEndpoints.equals(newPinnedEndpoints))
+        {
+            this.pinnedEndpoints = newPinnedEndpoints;
+
+            if (logger.isDebugEnabled())
+            {
+                logger.debug("Pinned "
+                    + Arrays.toString(pinnedEndpoints.toArray()));
+            }
+
+            firePropertyChange(PINNED_ENDPOINTS_PROPERTY_NAME,
```

Comment:
No changes expected, since this has been open for a long time and I never commented, but FYI: I've been trying to remove the usage of `PropertyChangeListener`, as it can be quite confusing (you have to do a string search to find usages, the value parameter is not typed, etc). I think an explicit method that Endpoint would override is a better option.